src/utils/DeploymentBuilder.ts

```diff
@@ -45,6 +35,42 @@ export class DeploymentBuilder {
         return { files: filesByHash, entityId: entity.id }
     }
 
+    /**
+     * In cases where we don't need to re-upload content files, we can simply generate the generate the new entity
+     */
+    static async buildEntityWithAlreadyUploadedHashes(type: EntityType, pointers: Pointer[], content: Map<string, ContentFileHash>, metadata?: EntityMetadata, timestamp?: Timestamp): Promise<DeploymentPreparationData> {
```

Comment:
This function contains a lot of duplicated code with `buildEntity`, could you implement one based on the other, or implement an internal function and call that function from this one and `buildEntity`?
We can not infer that both functions do pretty similar jobs without reading the code.